Sources/Logging/Logging.swift

```diff
@@ -472,24 +473,30 @@ public struct MultiplexLogHandler: LogHandler {
     public init(_ handlers: [LogHandler]) {
         assert(!handlers.isEmpty)
         self.handlers = handlers
+        self.effectiveLogLevel = MultiplexLogHandler.effectiveLogLevel(handlers.map { $0.logLevel })
     }
 
     public var logLevel: Logger.Level {
         get {
-            return self.handlers[0].logLevel
+            return self.effectiveLogLevel
         }
         set {
             self.mutatingForEachHandler {
                 $0.logLevel = newValue
             }
+            self.effectiveLogLevel = MultiplexLogHandler.effectiveLogLevel(self.handlers.map { $0.logLevel })
```

Comment:
since we are setting all the handlers in the move loop, computing effectiveLogLevel seems redundant

Reply:
that was "problem", example:

```
logger1 at .info
logger2 at .debug

multiplex.debug() // would never be emitted
```

because:

`multiplex.loglevel` was `return self.handlers[0].logLevel` arbitrary using the "first" as the handlers log level.

How logger decides if it should pass value to handler is: 

```
if handler.logLevel <= level {
```

so even though the `logger2` _does_ want to log things on debug level, it would not, because `logger1` is set to info; That's a bit weird.

----

Solutions:

- (this PR) on construction find "lowest loglevel this multiplex handler should accept, because one of its parts wants to get it"
- OR: make it like a normal logger/handler that it defaults to `.info` and if we want to log at other levels we need to change that on the multiplex handler?
- OR: (how it works today) the "first logger" is special and decides at which log level we're logging.
  - note that `multiplex.logLevel = .trace` then is a bit weird -- we've just changed _all_ loggers to .trace even though before we were saying the "first one only matters"; so the current impl is confusing since without setting a level: handlers log on VARIOUS levels; with setting a level, ALL handlers log at the same level.
  - rather what we'd want to do IMO when setting `multiplex.logLevel = .trace` is that "this handler will now take trace logs; and if the log handlers it contains WANT TO (were passed in as such) to log at .trace they now have the chance to do so" (this PR) -- so we retain the log level configurations that were passed in...
  - perhaps we'd then need a way to _really set_ the underlying levels if we wanted to? not sure.

Reply:
Oh wait, sorry didn't have my coffee yet! You mean that the looping here is redundant, thanks yes that's true.

☕ ☕ ☕